kore/src/Kore/Simplify/Predicate.hs

```
@@ -335,37 +335,51 @@ simplifyNot ::
+                    zipWith fromMaybe (map MultiOr.singleton ps) mbSimplifieds
+            pure (anySimplified, MultiAnd.make results)
+
+    pure $
+        if (or flags)
+            then fold . MultiOr.make $ map liftOrs andOrs
+            else normal
   where
-    applyUserDefined :: MaybeT simplifier NormalForm
-    applyUserDefined = do
-        -- produce a single termlike that we can use for matching
-        let arg :: TermLike RewritingVariableName
-            arg = OrPattern.toTermLike helpSort $ toOrPattern helpSort multiOr
-            -- HAAAACK: sort stripped in NormalForm of predicates
+    applyUserDefined ::
+        Predicate RewritingVariableName ->
+        MaybeT simplifier (MultiOr (Predicate RewritingVariableName))
+    applyUserDefined predicate = do
+        -- produce a termlike that we can use for matching
+        let -- HAAAACK: sort stripped in NormalForm of predicates
             helpSort = SortVariableSort . SortVariable $ noLocationId "SomeSort"
         -- call the equation matcher
         applied <-
             Axiom.evaluatePattern
                 sideCondition
                 Condition.top
-                (TermLike.mkNot arg)
+                (Predicate.fromPredicate helpSort predicate)
                 (const empty)
-        -- convert result back to NormalForm
+        -- convert result back to Predicate
         traverse_ (guard . Conditional.isPredicate) applied
-        pure $
-            MultiOr.map
-                (Predicate.toMultiAnd . from . snd . Conditional.splitTerm)
-                applied
-
-    standardSimplification :: simplifier NormalForm
-    standardSimplification = do
-        disjunctiveNormalForms <- Logic.observeAllT $ do
-            multiAnd <- Logic.scatter multiOr
-            lift $ normalizeNotAnd Not{notSort = (), notChild = multiAnd}
-        normalizeMultiAnd (MultiAnd.make disjunctiveNormalForms)
+        pure $ MultiOr.map (from . snd . Conditional.splitTerm) applied
```

We should probably throw an error if any of the results is not a predicate.

---

`traverse_ (guard . Conditional.isPredicate) applied` above takes care of that case. 
We can of course make the run fail hard instead of swallowing the error condition. Will modify accordingly